### ISSUE TITLE
Pokemon Sword/Shield random hangs

### ISSUE THREAD

**Issue by alessiot89** [yuzu-emu/yuzu#13119]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Mainline 1722

### Description of Issue

Randomly playing Pokemon Sword/Shield the game hangs but the audio (the music) continues to play. Most of time the issue is temporary fixed toggling fullscreen/windowed with F11. However, sometimes toggling between windowed and fullscreen mode doesn't fix and the game completely freezes. This happens both in windowed mode and in fullscreen mode.

### Expected Behavior

No hangs should occurs.

### Reproduction Steps

There is no specific way to reproduce it sadly, it happens randomly playing the game.

### Log File

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/14375083/yuzu_log.txt)

### System Configuration

CPU: Ryzen 5 3600X
GPU/Driver: RX 67000 XT driver "AMD Software: Adrenalin Edition 24.2.1 for HELLDIVERS™ 2" version 23.40.19.01
RAM: 32GB DDR4 @3600
OS: Windows 10 22H2 build 19045.3930
Controller: XBOX Series X/S controller, last firmware updated (the issue happens with both bluettoth and usb mode)

**Comment by alessiot89** [comment]
I think all this is related about controller polling rate, especially with BT devices... I noted when this happens than after few seconds the controller goes in idle and stops to responds to yuzu (the debug controller windows doesn't give signs of it being alive), fullscreen tabbing most of times "unstuck" the situation, like some kind of race condition maybe on controller polling being reset with the swap chain mode changes..

**Comment by Squall-Leonhart** [comment]
Issue is because of the dropping of the controller, all i can say is get a BT 4.1 receiver and nothing higher for the best compatibility with switch controllers.

**Comment by Squall-Leonhart** [comment]
> no, it's not a BT only issue, this happens also in wired mode.

Well yes, when disconnecting or connecting a controller the application will wait on the device setup within SDL's input handler, if  you try to interact with the window in this case windows itself may close the temporarily hung process, if the emulated game is communicating vibration or motion data to and from the controller at the time this disconnection happens you can wind up with an access violation.

> my bluetooth card is also an intel 9260, BT version: 5.1, which is working fine in every other scenarios outside yuzu and is more than enough to manage a 1-2 meters distance connection https://www.intel.com/content/www/us/en/products/sku/99445/intel-wirelessac-9260/specifications.html

5.1 receivers are wholly problematic with switch hardware, this is known fact.

**Comment by alessiot89** [comment]
the controller is not disconnecting. reconnecting the xbox series x/s controller takes some seconds in usb mode (like 3 or so) and even more in BT mode (like 10 or so), while toggling between fullscreen/windowed restores the input instantly.
The games even recognise the input I pressed while the controller wasn't working in-game, e.g.: the controller stops to responds in the game, last input pressed was the button mapped as "A" button, I toggle fullscreen/windowed, than the game immediately responds to the "A" button even if I remove my finger from it before toggling F11 key on my keyboard. Of course switching from USB to BT is a different task but I cannot

and this is happening pokemon sword (so I guess shield to since are basically the same game) even if before opening yuzu I turn off the BT card, I connect the controller with the usb cable and than I start yuzu and load the game.

so I guess it's some kind of SDL issue

I tried also to upgrade the xbox controller firmware (using the windows store xbox accessories app) but didn't help.

Which makes a bit harder to debug I guess, this happens randomly, it cannot for many minutes or it can happens frequently in few seconds. I am not familiar with yuzu debugging and profiling, but when this input hangs happens the rest of the game logic seems to continue fine, since graphics and music do not hang.. is there any profiling/debug view mode outside the controller debug window to see the input handling? like something from the polling rate or an input queue?

**Comment by Squall-Leonhart** [comment]
> edit: as for the usb bug on some AMD motherboards I didn't notice it, maybe my model isn't affected

It's a PCIE bus issue that can manifest when multiple gen4 devices are present on the GPU, the link on the cpu generates excessive errors and issues the usb controller a reset command, in some cases this reloads all hubs and devices but usually it acts like a dam has rapidly filled up and someone suddenly opens a flood gate letting forth all the buffered commands, it can be affected by altering the priority of Interrupts in the MSI Utility provided by guru3d user mbk1969

**Comment by alessiot89** [comment]
speaking about SDL, I also notice this, although not sure how much related: https://github.com/libsdl-org/SDL/issues/9106

I currently set the controller as "Pro Controller" since it matches better the xbox controller

**Comment by alessiot89** [comment]
seems like sdl 2.28.x may fix the problem https://github.com/Ryujinx/Ryujinx/issues/3301

**Comment by alessiot89** [comment]
> seems like sdl 2.28.2 may fix the problem [Ryujinx/Ryujinx#3301](https://github.com/Ryujinx/Ryujinx/issues/3301) https://github.com/libsdl-org/SDL/releases/tag/release-2.28.2
> 
> edit: dang, is yuzu already using it? :\

yes the version is the same however the binary it's not. not sure if SDL relese is wrong since all versions for windows are literally the same (same hashing), I downloaded the VC version (although as I just said it's totally equal to the mingw version) and the situation seems better. I need more time to test/play but I was able to have like over 1 hour session without random missing focus using the "official" SDL dll

**Comment by german77** [comment]
I doubt SDL is the issue here. Open the controller debugger... view->debugging->controller P1 you will see input in real time. If that stops working then is 100% sdl. If it's only in game then is an emulation bug.

**Comment by alessiot89** [comment]
When the controller stops to respond the controller debgger also show no input
so yes looks like an sdl issue
now I am using the official sdl dll and it seems fine.

**Comment by german77** [comment]
Only difference with official SDL binary is that we build without WGI support. SDL dies when yuzu is launched through steam and WGI is active.

**Comment by german77** [comment]
Yuzu currently is on 2.28.2. I can build one with 2.30.0 if that fixes the issues even without WGI. We can update the submodule.

**Comment by alessiot89** [comment]
> Only difference with official SDL binary is that we build without WGI support. SDL dies when yuzu is launched through steam and WGI is active.

I see however so far, a couple of hours or a bit more, using the official SDL 2.28.2 with WGI seems fine. I am not launching yuzu trough steam though.
